### PR TITLE
SWC-6283: Show selected version data in Dataset Editor, Entity Finder

### PR DESCRIPTION
**Pull request by nickgros**
- Dataset items editor will show data for the selected version of the dataset, such as annotations in the EntityBadgeIcons
- Entity Finder will show data for the selected version of the dataset, such as annotations in the EntityBadgeIcons
- Update DetailsView.test.tsx to reduce the amount of logged errors (mostly "not wrapped in act" warnings caused by not using RTL's `findBy*`)

## Screenshots

Entity Finder:

https://user-images.githubusercontent.com/17580037/213246141-53436432-004f-4931-9d89-615001890edc.mov

Dataset Editor:

https://user-images.githubusercontent.com/17580037/213246325-ecccc25c-55ad-42aa-af47-2512566f0747.mov

### CODE REVIEW

File: src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx [hunk 1]
```diff
@@ -72,8 +77,8 @@ const versionResult: PaginatedResults<VersionInfo> = {
   results: [
     {
       id: entityHeaders[0].id,
-      versionNumber: 5,
-      versionLabel: 'version 5 label',
+      versionNumber: 3,
+      versionLabel: 'version 3 label',
```
Comment (nickgros) [inline src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx:81]: The actual mock data has a `versionNumber` of 3. Having this conflict with what MSW was returning was causing issues.

File: src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx [hunk 2]
```diff
@@ -168,32 +173,35 @@ describe('DetailsView tests', () => {
     })
 
     describe('Determines correct row appearance', () => {
-      it('Creates a row with the default appearance', () => {
+      it('Creates a row with the default appearance', async () => {
```
Comment (nickgros) [inline src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx:176]: In many places, replace synchronous `getBy` with async `findBy` to get rid of many "not wrapped in act" warnings.

File: src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx [hunk 3]
```diff
@@ -712,6 +738,17 @@ describe('DetailsView tests', () => {
...
+        // The version number passed to renderers (such as the EntityBadgeIcons) should be undefined
+        await waitFor(() =>
+          expect(mockEntityBadgeIcons).toHaveBeenLastCalledWith(
+            expect.objectContaining({
+              entityId: entityHeaders[0].id,
+              versionNumber: undefined,
+            }),
+            expect.anything(),
+          ),
+        )
```
Comment (nickgros) [inline src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx:751]: New assertion for changed behavior.

File: src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx [hunk 4]
```diff
@@ -735,9 +772,20 @@ describe('DetailsView tests', () => {
...
+        // The version number passed to renderers (such as the EntityBadgeIcons) should be the selected version
+        await waitFor(() =>
+          expect(mockEntityBadgeIcons).toHaveBeenLastCalledWith(
+            expect.objectContaining({
+              entityId: entityHeaders[0].id,
+              versionNumber: versionResult.results[1].versionNumber,
+            }),
+            expect.anything(),
+          ),
+        )
```
Comment (nickgros) [inline src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx:785]: New assertion for changed behavior.

File: src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx [hunk 5]
```diff
@@ -762,6 +810,17 @@ describe('DetailsView tests', () => {
...
+        // The version number passed to renderers (such as the EntityBadgeIcons) should be the automatically selected version
+        await waitFor(() =>
+          expect(mockEntityBadgeIcons).toHaveBeenLastCalledWith(
+            expect.objectContaining({
+              entityId: entityHeaders[0].id,
+              versionNumber: versionResult.results[0].versionNumber,
+            }),
+            expect.anything(),
+          ),
+        )
```
Comment (nickgros) [inline src/__tests__/lib/containers/entity_finder/details/DetailsView.test.tsx:823]: New assertion for changed behavior.

File: src/__tests__/lib/containers/table/datasets/DatasetItemsEditor.test.tsx [hunk 6]
```diff
@@ -491,9 +495,31 @@ describe('Dataset Items Editor tests', () => {
...
+    await waitFor(() =>
+      expect(mockEntityBadgeIcons).toHaveBeenLastCalledWith(
+        expect.objectContaining({
+          entityId: mockFileReference.targetId,
+          versionNumber: 1,
+        }),
+        expect.anything(),
+      ),
+    )
+
```
Comment (nickgros) [inline src/__tests__/lib/containers/table/datasets/DatasetItemsEditor.test.tsx:522]: New assertions for changed behavior.

File: src/lib/containers/entity_finder/details/view/DetailsView.tsx [hunk 7]
```diff
@@ -250,16 +250,32 @@ export const DetailsView: React.FunctionComponent<DetailsViewProps> = ({
...
+            versionNumber = entity.versionNumber
+          }
+          // otherwise, show the current version's data (versionNumber is undefined)
+        }
+
         entities.push({
           ...entity,
           entityId: entity.id,
-          versionNumber:
-            'versionNumber' in entity ? entity.versionNumber : undefined,
```
Comment (nickgros) [inline src/lib/containers/entity_finder/details/view/DetailsView.tsx:257]: We used to just show information for the latest version number (if the entity was versionable)

File: src/lib/containers/table/datasets/DatasetItemsEditor.tsx [hunk 8]
```diff
@@ -461,7 +460,6 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     {
       key: 'status',
       width: 80,
-      dataKey: 'entityId',
```
Comment (nickgros) [inline src/lib/containers/table/datasets/DatasetItemsEditor.tsx:464]: For the Dataset Editor, the change is simple. By specifying this `dataKey`, we were omitting versionNumber passed to the `BadgeIconsRenderer` component. When we remove `dataKey`, the renderer gets all of the row data that was already present, including the `entityId` and `versionNumber`.

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**